drivers/mpu9150/mpu9150.c

```diff
@@ -0,0 +1,610 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     mpu9150
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the MPU-9150 9-Axis Motion Sensor
+ *
+ * @author      Fabian Nack <nack@inf.fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "mpu9150.h"
+#include "mpu9150-regs.h"
+#include "periph/i2c.h"
+#include "vtimer.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+#define REG_RESET           (0x00)
+#define MAX_VALUE           (0x7FFF)
+
+/* Default config settings */
+static const mpu9150_status_t DEFAULT_STATUS = {
+    .accel_pwr = PWR_ON,
+    .gyro_pwr = PWR_ON,
+    .compass_pwr = PWR_ON,
+    .gyro_fsr = GYRO_FSR_250DPS,
+    .accel_fsr = ACCEL_FSR_16G,
+    .sample_rate = 0,
+    .compass_sample_rate = 0,
+    .compass_x_adj = 0,
+    .compass_y_adj = 0,
+    .compass_z_adj = 0,
+};
+
+/* Internal function prototypes */
+static int compass_init(mpu9150_t *dev);
+static void conf_bypass(mpu9150_t *dev, uint8_t bypass_enable);
+static void conf_lpf(mpu9150_t *dev, uint16_t rate);
+
+/*---------------------------------------------------------------------------*
+ *                          MPU9150 Core API                                 *
+ *---------------------------------------------------------------------------*/
+
+int mpu9150_init(mpu9150_t *dev, i2c_t i2c, mpu9150_hw_addr_t hw_addr,
+        mpu9150_comp_addr_t comp_addr)
+{
+    char temp;
+
+    dev->i2c_dev = i2c;
+    dev->hw_addr = hw_addr;
+    dev->comp_addr = comp_addr;
+    dev->conf = DEFAULT_STATUS;
+
+    /* Initialize I2C interface */
+    if (i2c_init_master(dev->i2c_dev, I2C_SPEED_FAST)) {
+        DEBUG("[Error] I2C device not enabled\n");
+        return -1;
+    }
+
+    /* Acquire exclusive access */
+    i2c_acquire(dev->i2c_dev);
+
+    /* Reset MPU9150 registers and afterwards wake up the chip */
+    i2c_write_reg(dev->i2c_dev, dev->hw_addr, MPU9150_PWR_MGMT_1_REG, MPU9150_PWR_RESET);
+    vtimer_usleep(MPU9150_RESET_SLEEP_US);
```

Comment:
I'd propose to use `hwtimer_wait(HWTIMER_TICKS(MPU9150_RESET_SLEEP_US))` because of `vtimer`.